src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -56,21 +66,32 @@ ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
   // is made. It probably belongs where the key checking and
   // instantiateReactComponent is done.
 
-  var prevEmpty = prevElement === null || prevElement === false;
-  var nextEmpty = nextElement === null || nextElement === false;
+  var prevRef = null;
+  var prevOwner = null;
+  if (prevElement !== null && typeof prevElement === 'object') {
+    prevRef = prevElement.ref;
+    prevOwner = prevElement._owner;
+  }
+
+  var nextRef = null;
+  var nextOwner = null;
+  if (nextElement !== null && typeof nextElement === 'object') {
+    nextRef = nextElement.ref;
+    nextOwner = nextElement._owner;
+  }
 
   return (
-    // This has a few false positives w/r/t empty components.
-    prevEmpty || nextEmpty ||
-    nextElement.ref !== prevElement.ref ||
+    prevRef !== nextRef ||
     // If owner changes but we have an unchanged function ref, don't update refs
-    (typeof nextElement.ref === 'string' &&
-     nextElement._owner !== prevElement._owner)
+    (typeof nextRef === 'string' && nextOwner !== prevOwner)
```

Comment:
All the tests are passing but i'd love a second pair of eyes to know if this is safe